cli/src/git_util.rs

```diff
@@ -433,3 +456,82 @@ export or their "parent" bookmarks."#,
     }
     Ok(())
 }
+
+pub fn git_fetch(
+    ui: &mut Ui,
+    tx: &mut WorkspaceCommandTransaction,
+    settings: &UserSettings,
+    git_repo: &git2::Repository,
+    remotes: &[String],
+    branch: &[StringPattern],
+) -> Result<(), CommandError> {
+    for remote in remotes {
+        let stats = with_remote_git_callbacks(ui, None, |cb| {
+            git::fetch(
+                tx.repo_mut(),
+                git_repo,
+                remote,
+                branch,
+                cb,
+                &settings.git_settings(),
```

Comment:
nit: `settings` could be obtained from `tx.settings()` to reduce number of arguments.

Reply:
Ah nice. Done. Thanks.